guides/common/modules/proc_creating-a-custom-certificate.adoc

```diff
@@ -0,0 +1,85 @@
+[id="creating-a-custom-certificate_{context}"]
+
+= Creating a Custom SSL Certificate for {ProductName}
+
+ifeval::["{context}" == "{project-context}"]
+Use this procedure to create a custom SSL certificate for {ProductName}.
+If you already have a custom SSL certificate for {ProductName}, skip this procedure.
+endif::[]
+
+ifeval::["{context}" == "{smart-proxy-context}"]
+On {ProjectServer}, create a custom certificate for your {ProductName}.
+If you already have a custom SSL certificate for {ProductName}, skip this procedure.
+endif::[]
+
+.Procedure
+
+. To store all the source certificate files, create a directory that is accessible only to the `root` user.
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+# mkdir /root/{context}_cert
+----
+
+. Create a private key with which to sign the Certificate Signing Request (CSR).
++
+Note that the private key must be unencrypted.
+If you use a password-protected private key, remove the private key password.
++
+If you already have a private key for this {ProductName}, skip this step.
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+# openssl genrsa -out `/root/{context}_cert/{context}_cert_key.pem` 4096
+----
+
+. Create the `/root/{context}_cert/openssl.cnf` configuration file for the Certificate Signing Request (CSR) and include the following content:
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+[ req ]
+req_extensions = v3_req
+distinguished_name = req_distinguished_name
+x509_extensions = usr_cert
+prompt = no
+
+[ req_distinguished_name ]
+CN = _{context}.example.com_
+
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment
+extendedKeyUsage = serverAuth, clientAuth, codeSigning, emailProtection
+subjectAltName = @alt_names
+
+[ usr_cert ]
+basicConstraints=CA:FALSE
+nsCertType = client, server, email
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+extendedKeyUsage = serverAuth, clientAuth, codeSigning, emailProtection
+nsComment = "OpenSSL Generated Certificate"
+subjectKeyIdentifier=hash
+authorityKeyIdentifier=keyid,issuer
+
+[ alt_names ]
+DNS.1 = _{context}.example.com_
+----
+. Generate the Certificate Signing Request (CSR):
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+# openssl req -new \
+-key _/root/{context}_cert/{context}_cert_key.pem_ \ <1>
+-config _/root/{context}_cert/openssl.cnf_ \ <2>
+-out _/root/{context}_cert/{context}_cert_csr.pem_ <3>
+----
+<1> Path to the private key.
+<2> Path to the configuration file.
+<3> Path to the CSR to generate.
+
+. Send the certificate signing request to the Certificate Authority.
```

Comment:
This resembles `CA`, same as Certificate Signing Request (CSR). Maybe we can add `(CA)` in the suffix?

Reply:
CA is an acronym it so makes sense it is capitalized, however; I don't know about the full phrase. Opinions? @maximiliankolb @Lennonka @asteflova

Reply:
I agree with lower case, but yes to adding `(CA)` after that.

Reply:
This also applies to other modules but should have a separate PR.

Reply:
@mjivraja @Lennonka how about adding `(CA)` suffix to the first occurrence and then only use `CA` instead of `certificate authority (CA)`? Same things (lowercase, suffix, etc.) also apply to "subject alt name" , "common name", "certificate signing request", and "privacy-enhanced mail"

Reply:
I would not go for this. End-users may need to open the glossary often to get the value of acronyms.

Reply:
HTML has the [`<abbr>`-tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/abbr). There's an issue for asciidoctor from 2013 to support this, but sadly it's still open. https://github.com/asciidoctor/asciidoctor/issues/252 does suggest some custom macros to support it. Something to consider.

Reply:
> I would not go for this. End-users may need to open the glossary often to get the value of acronyms.

That is it would be explained on the first occurence.

Reply:
CA as an acronym for certificate authority is used so commonly that I believe we can use it without any worries (after writing it out on first occurrence, just in case).

Reply:
Actually, writing the full expression on the first occurence in the module (not the whole documentation) and then going just with the abbreviation in the rest of the module is sufficient. (Even the IBM style guide says so.)

Reply:
That's what I meant -- expand it for the first time in the module, then use it without worries. (Sorry if I didn't make myself clear enough :))